From 2584182f3896b459e941ce96e85760121ea0cf8f Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Mon, 15 Mar 2021 17:09:20 +0100
Subject: [PATCH] Fix warning from SDDM::generateName()

Properly constructs the created string by allocating the space to fill
instead of just reserving. Addresses the following warning:
   Using QCharRef with an index pointing outside the valid range of a QString.
   The corresponding behavior is deprecated, and will be changed in a future
   version of Qt.
---
 src/daemon/Utils.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/daemon/Utils.h b/src/daemon/Utils.h
index 7de5a98..03170f8 100644
--- a/src/daemon/Utils.h
+++ b/src/daemon/Utils.h
@@ -26,11 +26,11 @@
 namespace SDDM {
 
 inline QString generateName(int length) {
-    QString digits = QStringLiteral("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
+    const QString digits = QStringLiteral("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
 
     // reserve space for name
     QString name;
-    name.reserve(length);
+    name.resize(length);
 
     // create random device
     std::random_device rd;
-- 
2.33.0

